Updating the codeowners file

bbarani opened the pull request:
### Description
Updating the CODEOWNERS file to reflect the maintainers on MAINTAINERS.md file

### Issues Resolved
Closes: #235 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

bbarani commented:
CODEOWNERS file is not green yet. Couldn't find user @anshul291995. @Bukhtawar @eirsep Can you please confirm if this user needs to be added to this repo?

Bukhtawar commented:
Barani,  @anshul291995 is a former contributor to this repository, we can skip access for now

bbarani commented:
@Bukhtawar  @eirsep I have updated the MAINTAINERS list and removed @anshul291995 now. I notice some CI failures so need your help to remediate it before merging this PR.

dblock commented on MAINTAINERS.md:
```diff
@@ -7,5 +7,4 @@ This document contains a list of maintainers in this repo. See [opensearch-proje
 | --------------------- | ----------------------------------------------- | ----------- |
 | Bukhtawar Khan        | [Bukhtawar](https://github.com/Bukhtawar)       | Amazon      |
-| Surya Sashank Nistala | [eirsep](https://github.com/eirsep)             | Amazon      |
-| Anshul Agarwal        | [anshul291995](https://github.com/anshul291995) | Amazon      |
```
Let's move those folks to a new Emeritus section, don't remove them from here.
